src/index.ts

```diff
@@ -38,7 +38,7 @@ export interface HalLink {
 /**
  * A HAL document
  */
-export interface HalResource {
+export interface HalResource<T extends Record<string, any>> extends T {
```

Comment:
Great idea, but one question, can we also make `Record<string, any>` the default?

```suggestion
export interface HalResource<T extends Record<string, any> = Record<string, any>> extends T {
```